internal/fshasher/fshasher.go

```diff
@@ -114,9 +114,10 @@ func writeFile(ctx context.Context, w io.Writer, f fs.File) error {
 	if err != nil {
 		return err
 	}
-	defer r.Close()
 
-	_, err = io.Copy(w, r)
+	if _, err = io.Copy(w, r); err != nil {
+		return err
+	}
 
-	return err
+	return r.Close()
```

Comment:
This actually changes the behavior and would leave `r` open when `io.Copy` fails.

Reply:
fixed

Reply:
one sec, let me fix it again :)

Reply:
done, I initially misread the code and thought that it's the `w` that needs to be closed, simply adding `//nolint:errcheck` is the right thing to do for read-only streams